yarn-project/pxe/src/pxe_service/pxe_service.ts

```diff
@@ -214,16 +271,181 @@ export class PXEService implements PXE {
     return this.jobQueue.put(fn);
   }
 
-  isL1ToL2MessageSynced(l1ToL2Message: Fr): Promise<boolean> {
-    return this.node.isL1ToL2MessageSynced(l1ToL2Message);
+  async #registerProtocolContracts() {
+    const registered: Record<string, string> = {};
+    for (const name of protocolContractNames) {
+      const { address, contractClass, instance, artifact } =
+        await this.protocolContractsProvider.getProtocolContractArtifact(name);
+      await this.contractDataProvider.addContractArtifact(contractClass.id, artifact);
+      await this.contractDataProvider.addContractInstance(instance);
+      registered[name] = address.toString();
+    }
+    this.log.verbose(`Registered protocol contracts in pxe`, registered);
+  }
+
+  async #isContractClassPubliclyRegistered(id: Fr): Promise<boolean> {
+    return !!(await this.node.getContractClass(id));
+  }
+
+  async #isContractPubliclyDeployed(address: AztecAddress): Promise<boolean> {
+    return !!(await this.node.getContract(address));
+  }
+
+  async #isContractInitialized(address: AztecAddress): Promise<boolean> {
+    const initNullifier = await siloNullifier(address, address.toField());
+    return !!(await this.node.getNullifierMembershipWitness('latest', initNullifier));
+  }
+
+  /**
+   * Retrieves the simulation parameters required to run an ACIR simulation.
+   * This includes the contract address, function artifact, and historical tree roots.
+   *
+   * @param execRequest - The transaction request object containing details of the contract call.
+   * @returns An object containing the contract address, function artifact, and historical tree roots.
```

Comment:
I don't see roots and artifacts in the return object.

Reply:
This function is unused, got a bad merge from Adam's PR. Removed!